### PR TITLE
[Review][Speech] Add Speech Framework from Xcode 8 Beta 1

### PR DESCRIPTION
**Pull request by dalexsoto**
FIXME: `SFSpeechRecordingRecognitionRequest` is not in the public api
filled radar://26799291 https://trello.com/c/s6s6YKua

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by spouliot** [comment]
That also include CallKit (previous commit/PR ?) and they should be committed separately.

File: src/speech.cs [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+		void DidHypothesizeTranscription (SFSpeechRecognitionTask task, SFTranscription transcription);
+
+		[Export ("speechRecognitionTask:didFinishRecognition:")]
+		void DidFinishRecognition (SFSpeechRecognitionTask task, SFSpeechRecognitionResult recognitionResult);
+
+		[Export ("speechRecognitionTaskFinishedReadingAudio:")]
+		void FinishedReadingAudio (SFSpeechRecognitionTask task);
+
+		[Export ("speechRecognitionTaskWasCancelled:")]
+		void SpeechRecognitionTaskWasCancelled (SFSpeechRecognitionTask task);
```
Comment (spouliot) [inline src/speech.cs]: `TaskWasCancelled` (like `DidDetectSpeech` above)

File: src/speech.cs [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+		[Export ("defaultTaskHint", ArgumentSemantic.Assign)]
+		SFSpeechRecognitionTaskHint DefaultTaskHint { get; set; }
+
+		// FIXME: SFSpeechRecordingRecognitionRequest is not in the public api
+		// filled radar://26799291 
+		//[Export ("prepareWithRequest:")]
+		//void Prepare (SFSpeechRecordingRecognitionRequest request);
+
+		[Export ("recognitionTaskWithRequest:resultHandler:")]
+		SFSpeechRecognitionTask GetRecognitionTask (SFSpeechRecognitionRequest request, Action<SFSpeechRecognitionResult, NSError> resultHandler);
```
Comment (spouliot) [inline src/speech.cs]: candidate for `[Async]` ?

**Comment by dalexsoto** [comment]
@spouliot Yes, I worked this on top of my `xcode8` branch which includes CallKit. Once https://github.com/xamarin/xamarin-macios/pull/189 is merged this should not happen again 😛

🐻 with 🤓 this time 😃

**Comment by dalexsoto** [comment]
build

File: src/speech.cs (hunk 1 above)
Comment (dalexsoto) [inline src/speech.cs]: Fixed

File: src/speech.cs (hunk 2 above)
Comment (dalexsoto) [inline src/speech.cs]: Fixed

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by dalexsoto** [comment]
Fixing in another PR CallKit being on the same PR

**Pull request closed without merging**